Editorialize greasing considerations (#15)
intarchboard/draft-protocol-greasing: draft-edm-protocol-greasing.md

Greasing can take many forms, depending on the protocol and the nature of its
extension points.

Many protocols register values, codepoints, or numbers in a limited space. A

> common approach is to reserve a subset of the space for greasing (see

Reviewer: ```suggestion
common approach that has developed in more recent protocols is to reserve a subset of the space for greasing (see
```

Author: good qualification

as individually special. Instead of identifying each grease value explicitly,
it is better to have a "catch all" mechanism that can handle receipt of unknown
extensions, whether grease values or not, gracefully or without error.

It is recommended that senders pick an unpredictable grease value to include in
relevant protocol elements. This ensures that receiver greasing requirements are
exercised. Using predictable grease values risks ossification. To increase the
variety of grease values, it is advised to reserve a large range. However, the
specific size and distribution of the grease range needs to accommodate the
protocol constraints. For instance, protocols that use 8-bit fields may find it
too costly to dedicate many grease values, while 32-bit or 64-bit fields are
likely to have no limitations.

It is recommended that senders use grease values at unpredictable times or
sequence points during protocol interactions. This avoids receivers
unintentionally ossifying on the occurrence of greasing in the temporal or
spatial domain.

It is recommended that large grease value sets are allocated in protocol

> documents by defining a unique algorithm, this increases the chance that

Reviewer: ```suggestion
documents by defining a unique algorithm, to increase the chance that
```

receiver greasing requirements are exercised. However, the choice of algorithm
needs to consider the spread of values and the size of contiguous blocks between
grease values. It is common for protocol extension designers to want to reserve
a contiguous block of code points in order to aid iteration and experimentation.
Small contiguous blocks increase the chance that such reservations might
unintentionally use grease values, which could lead to interoperability
failures.

> Fields that use grease values should request that IANA requests allocate only a

Reviewer: ```suggestion
Fields that use greased values should request that IANA requests allocate only a
```
Is this right?

Reviewer: Or greasing?

Author: I was attempting keep keep consistency with

> Values
reserved for the purpose of greasing are herein referred to as grease values.

 but really, I don't think "fields that x" can request anything from IANA because they are inanimate objects :joy:. Will take another stab.

single registry entry for the entire grease value set. When an algorithm has
been used, it should be included in the entry; see for example
https://www.iana.org/assignments/http3-parameters/http3-parameters.xhtml#http3-parameters-frame-types.

> Values in the grease value set must not be used or registered for any other

Reviewer: Grease/greased/greasing?
